packages/snaps-simulator/src/contexts/SnapInterface.tsx

```diff
@@ -0,0 +1,202 @@
+
+  /**
+   * Send a request to the Snap.
+   *
+   * @param event - The event type.
+   * @param name - The name of the component emitting the event.
+   * @param value - The value of the component emitting the event.
+   */
+  const rawSnapRequestFunction = (
+    event: UserInputEventType,
+    name?: string,
+    value?: string | InterfaceState,
+  ) => {
+    dispatch(
+      sendRequest({
+        origin: '',
+        handler: HandlerType.OnUserInput,
+        request: {
+          jsonrpc: '2.0',
+          method: ' ',
+          params: {
+            event: {
+              type: event,
+              // TODO: Allow null in the types and simplify this
+              ...(name !== undefined && name !== null ? { name } : {}),
+              ...(value !== undefined && value !== null ? { value } : {}),
+            },
+            id,
+          },
+        },
+      }),
+    );
+  };
+
+  /**
+   * Debounce the request to the Snap.
+   */
+  const snapRequestDebounced = debounce(rawSnapRequestFunction, 200);
+
+  /**
+   * Throttle the request to the Snap.
+   */
+  const snapRequestThrottled = throttle(rawSnapRequestFunction, 200);
+
+  /**
+   * Handle the submission of an user input event to the Snap.
+   *
+   * @param options - An options bag.
+   * @param options.event - The event type.
+   * @param options.name - The name of the component emitting the event.
+   * @param options.value - The value of the component emitting the event.
+   */
+  const handleEvent: HandleEvent = ({
+    event,
+    name,
+    value = name ? state[name] ?? undefined : undefined,
+  }) => {
+    const fn = THROTTLED_EVENTS.includes(event)
+      ? snapRequestThrottled
+      : snapRequestDebounced;
+    fn(event, name, value);
+  };
+
+  const getValue: GetValue = (name, form) => {
+    const value = form
+      ? (state[form] as FormState)?.[name]
+      : (state as FormState)?.[name];
+
+    if (value) {
+      return value;
+    }
+
+    return undefined;
+  };
+
+  /**
+   * Handle the change of an input value.
+   *
+   * @param name - The name of the input.
+   * @param value - The new value of the input.
+   * @param form - The form containing the input.
+   */
+  const handleInputChange: HandleInputChange = (name, value, form) => {
+    const newState = mergeValue(state, name, value, form);
+
+    snapInterfaceController?.updateInterfaceState(id, newState);
+    dispatch(setSnapInterfaceState(newState));
+
+    handleEvent({
+      event: UserInputEventType.InputChangeEvent,
+      name,
+      value: value ?? '',
+    });
+  };
+
+  return (
+    <SnapInterfaceContext.Provider
+      value={{ getValue, handleInputChange, handleEvent }}
+    >
+      {children}
+    </SnapInterfaceContext.Provider>
+  );
+};
+
+/**
+ * Get the Snap Interface context.
+ *
+ * @returns The Snap Interface context.
+ */
+export function useSnapInterfaceContext() {
+  return useContext(SnapInterfaceContext) as SnapInterfaceContextType;
```

Comment:
Maybe create an initial context with no-op functions instead of null, to avoid the type cast here.